stlab/concurrency/default_executor.hpp

```diff
@@ -55,9 +56,33 @@ namespace detail {
 
 /**************************************************************************************************/
 
+enum class executor_priority
+{
+    high,
+    medium,
+    low
+};
+
+/**************************************************************************************************/
+
 #if STLAB_TASK_SYSTEM == STLAB_TASK_SYSTEM_LIBDISPATCH
 
-struct default_executor_type {
+constexpr auto platform_priority(executor_priority p)
+{
+    switch (p)
+    {
+        case executor_priority::high:
+            return DISPATCH_QUEUE_PRIORITY_HIGH;
+        case executor_priority::medium:
+          return DISPATCH_QUEUE_PRIORITY_DEFAULT;
+        default:
```

Comment:
To be very picky here, I'd rather see a `switch` statement that covers all `executor_priority` cases instead of having a `default` case. Then, in the event a new executor priority is added, we get a compiler warning/error instead of a bug when the switch defaults.